python/lsst/ts/scheduler/utils/fbs_utils.py

```diff
@@ -0,0 +1,167 @@
+
+__all__ = [
+    "SchemaConverter",
+]
+
+import sqlite3
+
+import pandas as pd
+import numpy as np
+
+from rubin_sim.scheduler.utils import empty_observation
+
+
+class SchemaConverter:
+    """Record how to convert an observation array to the standard
+    opsim schema.
+    """
+
+    def __init__(self) -> None:
+        # Conversion dictionary, keys are opsim schema, values are
+        # observation dtype names
+        self.convert_dict = {
+            "observationId": "ID",
+            "night": "night",
+            "observationStartMJD": "mjd",
+            "observationStartLST": "lmst",
+            "numExposures": "nexp",
+            "visitTime": "visittime",
+            "visitExposureTime": "exptime",
+            "proposalId": "survey_id",
+            "fieldId": "field_id",
+            "fieldRA": "RA",
+            "fieldDec": "dec",
+            "altitude": "alt",
+            "azimuth": "az",
+            "filter": "filter",
+            "airmass": "airmass",
+            "skyBrightness": "skybrightness",
+            "cloud": "clouds",
+            "seeingFwhm500": "FWHM_500",
+            "seeingFwhmGeom": "FWHM_geometric",
+            "seeingFwhmEff": "FWHMeff",
+            "fiveSigmaDepth": "fivesigmadepth",
+            "slewTime": "slewtime",
+            "slewDistance": "slewdist",
+            "paraAngle": "pa",
+            "rotTelPos": "rotTelPos",
+            "rotTelPos_backup": "rotTelPos_backup",
+            "rotSkyPos": "rotSkyPos",
+            "rotSkyPos_desired": "rotSkyPos_desired",
+            "moonRA": "moonRA",
+            "moonDec": "moonDec",
+            "moonAlt": "moonAlt",
+            "moonAz": "moonAz",
+            "moonDistance": "moonDist",
+            "moonPhase": "moonPhase",
+            "sunAlt": "sunAlt",
+            "sunAz": "sunAz",
+            "solarElong": "solarElong",
+            "note": "note",
+        }
+        # Column(s) not bothering to remap:  'observationStartTime': None,
+        self.inv_map = {v: k for k, v in self.convert_dict.items()}
+        # angles to convert
+        self.angles_rad2deg = [
+            "fieldRA",
+            "fieldDec",
+            "altitude",
+            "azimuth",
+            "slewDistance",
+            "paraAngle",
+            "rotTelPos",
+            "rotSkyPos",
+            "rotSkyPos_desired",
+            "rotTelPos_backup",
+            "moonRA",
+            "moonDec",
+            "moonAlt",
+            "moonAz",
+            "moonDistance",
+            "sunAlt",
+            "sunAz",
+            "sunRA",
+            "sunDec",
+            "solarElong",
+            "cummTelAz",
+        ]
+        # Put LMST into degrees too
+        self.angles_hours2deg = ["observationStartLST"]
+
+    def obs2opsim(
+        self,
+        obs_array: np.ndarray,
+        filename: str,
+    ) -> None:
+        """Convert an array of observations into a pandas dataframe
+        with Opsim schema.
```

Comment:
Please also describe that the dataframe optionally gets stored in an sqlite database.